js_modules/dagster-ui/packages/ui-core/src/runs/RunUtils.tsx

```diff
@@ -107,14 +111,99 @@ export async function handleLaunchResult(
 
     if ('errors' in result) {
       message += ` Please fix the following errors:\n\n${result.errors
-        .map((error) => error.message)
+        .map((error: {message: any}) => error.message)
         .join('\n\n')}`;
     }
 
     showCustomAlert({body: message});
   }
 }
 
+export async function handleLaunchMultipleResult(
+  result: void | null | LaunchMultipleRunsMutation['launchMultipleRuns'],
+  history: History<unknown>,
+  options: {behavior: LaunchBehavior; preserveQuerystring?: boolean},
+) {
+  if (!result) {
+    showCustomAlert({body: `No data was returned. Did dagster-webserver crash?`});
+    return;
+  }
+  const successfulRunIds: string[] = [];
+  const failedRunsErrors: {message: string}[] = [];
+
+  if (result.__typename === 'PythonError') {
+    // if launch multiple runs errors out, show the PythonError and return
+    showCustomAlert({
+      title: 'Error',
+      body: <PythonErrorInfo error={result} />,
+    });
+    return;
+  } else if (result.__typename === 'LaunchMultipleRunsResult') {
+    // show corresponding toasts
+    const launchMultipleRunsResult = result.launchMultipleRunsResult;
+
+    for (const individualResult of launchMultipleRunsResult) {
+      if (individualResult.__typename === 'LaunchRunSuccess') {
+        successfulRunIds.push(individualResult.run.id);
+
+        const pathname = `/runs/${individualResult.run.id}`;
+        const search = options.preserveQuerystring ? history.location.search : '';
+        const openInSameTab = () => history.push({pathname, search});
+
+        // using open with multiple runs will spam new tabs
+        if (options.behavior === 'open') {
+          openInSameTab();
+        }
+        document.dispatchEvent(new CustomEvent('run-launched'));
```

Comment:
lets dispatch this event after the for loop so that we only do it once rather than multiple times.